src/components/AnimateSharedLayout/stack.ts

```diff
@@ -210,17 +205,29 @@ export class LayoutStack {
         return this.lead && this.lead?.presence !== Presence.Exiting
     }
 
-    shouldStackAnimate() {
+    shouldStackAnimate(): boolean {
+        if (this.lead && this.lead?.isPresent)
+            return this.shouldPrimaryAnimate(this.lead, this.follow)
+
+        if (this.follow)
+            return this.shouldPrimaryAnimate(this.follow, this.lead)
+
         return true
-        // return this.lead && this.lead?.isPresent
-        //     ? this.lead?.props?._shouldAnimate === true
-        //     : this.follow && this.follow?.props._shouldAnimate === true
+    }
+
+    shouldPrimaryAnimate(
+        primary: HTMLVisualElement,
+        secondary?: HTMLVisualElement
+    ) {
+        if (primary.config._shouldAnimateLayout === undefined) return true
+        secondary && secondary.config._shouldAnimateLayout?.(false)
+        return primary.config._shouldAnimateLayout(true) as boolean
     }
 
     getFollowOrigin(): AxisBox2D | undefined {
         // This shouldAnimate check is quite specifically a fix for the optimisation made in Framer
         // where components are kept in the tree ready to be re-used
-        return this.follow // && this.follow.shouldAnimate
+        return this.follow
```

Comment:
I think I removed the commented flag in a PR after `main` was forked to fix a framer bug